### PR TITLE
Installs Enterprise Splunk 8.0.1

### PR DESCRIPTION
**Pull request by haidangwa**
### Description

    - Installs Enterprise Splunk 8.0.1
    - Installs Splunk Universal Forwarder 8.0.1
    - Removes the dependency on chef-vault, relying purely on encrypted data bags
    - in `chef-splunk::client`, do not install the forwarder if the server config is being installed
      on a splunkd server
    - adds helper method: `#server?` that will return true if the chef node is a splunkd server
    - adds helper method: `#port_open?` that will return true if the local node has a specified port open
    - logs a warning message if `node['splunk']['upgrade']['server_url']` or `node['splunk']['upgrade']['forwarder_url']` exist
    

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -90,18 +91,26 @@
 default['splunk']['splunk_servers'] = []
 
 default['splunk']['forwarder'] = {
-  'url' => value_for_platform_family(
-    %w(rhel fedora suse amazon) => 'https://download.splunk.com/products/universalforwarder/releases/6.6.0/linux/splunkforwarder-6.6.0-1c4f3bbe1aea-linux-2.6-x86_64.rpm',
-    ['debian'] => 'https://download.splunk.com/products/universalforwarder/releases/6.6.0/linux/splunkforwarder-6.6.0-1c4f3bbe1aea-linux-2.6-amd64.deb'
+  'url' => value_for_platform(
```
Comment (tas50) [inline attributes/default.rb]: You probably just want to use value_for_platform_family here which would catch the derivatives and wouldn't require the 'default' hash key.

File: recipes/upgrade.rb [hunk 2]
```diff
@@ -16,15 +16,31 @@
...
+  msg = "DEPRECATED: node['splunk']['upgrade']['server_url'] was found. This attribute will be removed in a future version. " \
+        "Please use node['splunk']['server']['upgrade']['url'] instead"
+  log msg do
+    level :warn
+  end
+end
+
+if node['splunk'].attribute?('upgrade') && node['splunk']['upgrade'].attribute?('forwarder_url')
+  msg = "DEPRECATED: node['splunk']['upgrade']['forwarder_url'] was found. This attribute will be removed in a future version. " \
+        "Please use node['splunk']['forwarder']['upgrade']['url'] instead"
```
Comment (tas50) [inline recipes/upgrade.rb]: very nice!

File: attributes/default.rb (hunk 1 above)
Comment (haidangwa) [reply attributes/default.rb]: Correct me if I'm wrong, Chef doesn't publish what derivatives fall under what platform family, but it is documented what each platform 'String' should be.

File: attributes/default.rb [hunk 3]
```diff
@@ -91,17 +92,17 @@
 
 default['splunk']['forwarder'] = {
   'url' => value_for_platform_family(
-    %w(rhel fedora suse amazon) => 'https://download.splunk.com/products/universalforwarder/releases/6.6.0/linux/splunkforwarder-6.6.0-1c4f3bbe1aea-linux-2.6-x86_64.rpm',
```
Comment (tas50) [inline attributes/default.rb]: Add amazon back and merge away

**Pull request merged**